src/main/java/org/opentripplanner/transit/service/DefaultTransitService.java

```diff
@@ -577,4 +605,20 @@ public TransferService getTransferService() {
   public boolean transitFeedCovers(Instant dateTime) {
     return transitModel.transitFeedCovers(dateTime);
   }
+
+  /**
+   * Take a stream of T, count the occurrences of each value and return it in order of frequency
+   * from high to low.
+   * <p>
+   * Example: [a,b,b,c,c,c] will return [c,b,a]
+   */
+  private static <T> Stream<T> sortByOccurenceAndReduce(Stream<T> input) {
+    return input
+      .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))
+      .entrySet()
+      .stream()
+      // a bit tricky to read: reverse the order so that the higher numbers come first
+      .sorted(Comparator.comparing(entry -> -entry.getValue()))
```

Comment:
Would using `.reversed()` be an option on the `Comparator`?

Reply:
I used `reversed` but I had to add type arguments and I'm not sure why: 91225b70d47c0824cbcc0aee5ddda8164b9eb960

I think the result nevertheless more readable.